### PR TITLE
[MM-44887] Consistency in dates between the various payment modals

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
These screens would always show the first of the next month, which isn't necessarily the next billing date for this subscription. This changes it to show the trial end date if there is one (and it's in the future), otherwise it will show today's date. Essentially, this now communicates at what point their usage will count towards billing, rather than when the next billing cycle date is. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44887

#### Related Pull Requests

- Has server changes (please link here)

#### Screenshots
Trial End Jul 9:
image

#### Release Note

```release-note
None
```

### CODE REVIEW

**Comment by marianunez** [comment]
> These screens would always show the first of the next month, which isn't necessarily the next billing date for this subscription. This changes it to show the trial end date if there is one (and it's in the future), otherwise it will show today's date. Essentially, this now communicates at what point their usage will count towards billing, rather than when the next billing cycle date is.
> 

`Payment begins` I think we had intended to say when they will start `Paying` that's why we had it the first of the next month. It seems that now we are changing the meaning to say when they start incurring charges? Thought on this cc/ @johndavidlugtu

**Comment by johndavidlugtu** [comment]
@marianunez yeah so @nickmisasi brought this up earlier with me and I had signed off on it. **But** I had the chance of thinking it through some more and here's my thought process:

- I saw the appeal to stating when they start incurring charges, but there isn't really a neat way to describe that. "Payment begins" can be interpreted as when they are required to pay which isn't true in this case. Their first payment is due once the bill has been calculated.

- "Plan begins" sound like the best fit to describe when they start incurring charges, but the trial flow prevents us from using this phrase. Once a workspace in trial upgrades, their plan switches to paid immediately, but they won't incur charges until after the plan. So saying "Plan begins {when trial is over}" is incorrect

I think the best path forward here is to change the copy to say: "1st bill due: {date when first bill is due}"

**Comment by marianunez** [comment]
> @marianunez yeah so @nickmisasi brought this up earlier with me and I had signed off on it. **But** I had the chance of thinking it through some more and here's my thought process:
> 
> * I saw the appeal to stating when they start incurring charges, but there isn't really a neat way to describe that. "Payment begins" can be interpreted as when they are required to pay which isn't true in this case. Their first payment is due once the bill has been calculated.
> * "Plan begins" sound like the best fit to describe when they start incurring charges, but the trial flow prevents us from using this phrase. Once a workspace in trial upgrades, their plan switches to paid immediately, but they won't incur charges until after the plan. So saying "Plan begins {when trial is over}" is incorrect
> 
> I think the best path forward here is to change the copy to say: "1st bill due: {date when first bill is due}"
> 
> Apologies for the change back!

Agreed `Bill due` may be more accurate. @justinegeffen any feedback on this copy and the comments above ?

**Comment by justinegeffen** [comment]
If we're referring to "the day they need to make payment" then "First payment due" would work. I am not opposed to "First bill due" but payment and bill are different (imo) so that should be clarified - is this when they get the bill or when their payment is due? 

If we mean "the day from which billing starts" we could say "You'll be billed from".

**Comment by johndavidlugtu** [comment]
@justinegeffen it's when their bill is due and since we have their credit card on file, payment happens right away.

**Comment by johndavidlugtu** [comment]
@nickmisasi does that work? Let us know if the phrase is too long for the current design

**Comment by AGMETEOR** [comment]
> > @justinegeffen it's when their bill is due and since we have their credit card on file, payment happens right away.
> 
> Great. I think "You'll be billed from" is more accurate as it indicates the date from which they can expect to be charged. "Bill due" can imply that they have a choice whether to pay before that date or maybe even after. But if we bill them immediately I'd say "You'll be billed from" works best.

@justinegeffen How about "Paid usage starts: ...."?

File: utils/utils.tsx [hunk 1]
```diff
@@ -1716,6 +1717,16 @@ export function getNextBillingDate() {
     return nextBillingDate.format('MMM D, YYYY');
```
Comment (AGMETEOR) [inline utils/utils.tsx:1717]: With this change, I think `getNextBillingDate` is not used any where else. Maybe we can remove it.

**Review by AGMETEOR — APPROVED** [review]
Just left some small non-blocking comments. Otw, LGTM

**Comment by justinegeffen** [comment]
> @justinegeffen How about "Paid usage starts: ...."?

That works - another option is "Paid plan starts"? But that doesn't explicitly suggest that they'll be billed when that happens. That said, I can make sure the docs explain that or we could include hover text to explain it?

**Review by marianunez — APPROVED** [review]

**Comment by johndavidlugtu** [comment]
> Where does the "See how billing works" link go to, @johndavidlugtu? That could be a good place to add a more detailed explanation of timing etc.

It goes here @justinegeffen : https://docs.mattermost.com/manage/cloud-billing.html

**Pull request merged**